nemo_rl/data/multimodal_utils.py

```
@@ -179,3 +183,30 @@ def get_dim_to_pack_along(processor, key: str) -> int:
+
+    image_path can be either:
+    - path to local file
+    - url to image
+    - base64 encoded image
+    """
+    if isinstance(image_path_or_image, Image.Image):
+        return image_path_or_image
+
+    if image_path_or_image.startswith(("http://", "https://")):
+        # Handle URL
+        response = requests.get(image_path_or_image)
```

Ruff (0.14.13)

Probable use of `requests` call without timeout

(S113)